### PR TITLE
Add error handling for logical device creation

### PR DESCRIPTION
**Pull request by msiglreith**
Add error handling for `PhysicalDevice::open` using the Vulkan return codes.
Early feedback appreciated!

~~**[WIP]** Implement remaining backends~~

### CODE REVIEW

File: src/backend/vulkan/src/result.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+macro_rules! from_error {
+    { $($name:ident { $($base_error:ident => $err:ident,)* },)* } => {
+        $(
+            impl From<Error> for $name {
+                fn from(err: Error) -> Self {
+                    match err {
+                        $(
+                            Error::$base_error => $name::$err,
+                        )*
+                        _ => unreachable!("Unexpected error code ({:?}). Non specification conformant driver.", err),
```
Comment (msiglreith) [inline src/backend/vulkan/src/result.rs]: This part is nagging on me a bit, as I don't have experience on how much drivers do things by the spec regarding return codes..

Comment (kvark) [reply src/backend/vulkan/src/result.rs]: Agree. Can we have some sort of "unrecognized" error instead with logging the case to `error!`?

Comment (msiglreith) [reply src/backend/vulkan/src/result.rs]: Addind an `Unrecognized` variant to all error structs?

Comment (kvark) [reply src/backend/vulkan/src/result.rs]: Nah, on a second thought, we should be respecting the spec.

**Pull request merged**